src/translator/html.h

```diff
@@ -18,40 +21,84 @@ class BadHTML : public std::runtime_error {
 
 class HTML {
  public:
+  struct Options {
+    // List of elements for which we do not expect a closing tag, or self-closing
+    // elements in XHTML. See also https://developer.mozilla.org/en-US/docs/Glossary/Empty_element
+    // More relevant source of this list:
+    // https://searchfox.org/mozilla-central/rev/7d17fd1fe9f0005a2fb19e5d53da4741b06a98ba/dom/base/FragmentOrElement.cpp#1791
+    std::unordered_set<std::string> voidTags{"area",  "base",  "basefont", "bgsound", "br",    "col",
+                                             "embed", "frame", "hr",       "img",     "input", "keygen",
+                                             "link",  "meta",  "param",    "source",  "track", "wbr"};
+
+    std::unordered_set<std::string> inlineTags{"abbr",   "a", "b",    "em",    "i",    "kbd",    "mark", "math",
+                                               "output", "q", "ruby", "small", "span", "strong", "sub",  "sup",
+                                               "time",   "u", "var",  "wbr",   "ins",  "del",    "img"};
+
+    // List of characters that occur at the start of a token that indicate that
+    // the this token is probably *not* a continuation of a word. Set to empty
+    // to never mark a token as a continuation of the word.
+    // std::string continuationDelimiters = "\n ,.(){}[]";
+    std::string continuationDelimiters;
+
+    // Should we always add spaces to the places where tags used to be? I.e.
+    // `un<u>der</u>line` should become `un der line`?
+    bool substituteInlineTagsWithSpaces = true;
```

Comment:
make this part of response options? I feel defaulting to false is safer. Wouldn't mind making the whole `Options` part of `ResponseOptions`. The experiments you're doing by trial and error can be automated on a volume of HTML by means of crawling. WebAssembly via nodejs still looks primitive for the use-case. No clue about wasm-runtime.

Reply:
Just went and checked test-bench. I believe https://github.com/browsermt/bergamot-translator/commit/5dfeef874422f6dd4e5f4b482de66ef1dea50f53 should be beneficial to more people and a resolution here as well?

Reply:
I'm ok with putting the knobs in main even if just there to help development.  Doubt exposing them all the way through the browser is worth it.

Reply:
Oh btw I noticed Google Translate also has this behaviour of breaking words at tags, even more aggressively. I suspect they do this for a reason (which was my main reason for "let's go with *on* by default).

It aso doesn't seem to work all that well. For the `<u>d</u><i>o</i><b>g</b>` example it is a solution. But I noticed some of our translation models are really good at fixing mistakes with spaces: Both `transatlantic` and `trans atlantic` are often (correctly) translated as a single word. (Only `trans<br>atlantic` is successful in forcing the split in the translations as well.)

Reply:
1. Let's not block this on evaluations. We'll decouple it, give it a lower priority and give you more room to experiment.
2. Let's go with the Google default. 

@jelmervdl you know the tooling I have and how the said tooling can make things configurable. If you deem something should be good enough to give a knob outside for python-based things, please do add along with this PR. Since the value from bindings would be 1, we can look into this perhaps when the flags are more mature and you understand it well.